jsonapi/data_structs.go

```diff
@@ -64,7 +73,7 @@ type Data struct {
 	ID            string                  `json:"id"`
 	Attributes    json.RawMessage         `json:"attributes"`
 	Relationships map[string]Relationship `json:"relationships,omitempty"`
-	Links         *Links                  `json:"links,omitempty"`
+	Links         map[string]interface{}  `json:"links,omitempty"`
```

Comment:
this could directly be a `map[string]CustomLink` if the marshaler/unmarshaler knows when to produce a simple string instead of an object, right?